examples/helloworld/client.ts

```diff
@@ -496,4 +499,123 @@ export class HelloWorldAppClient {
     return this.call(HelloWorldAppCallFactory.helloWorldCheck(args, params))
   }
 
+  public compose(): HelloWorldAppComposer {
+    const client = this
+    const atc = new AtomicTransactionComposer()
+    let promiseChain:Promise<unknown> = Promise.resolve()
+    const resultMappers: Array<undefined | ((x: any) => any)> = []
+    return {
+      hello(args: MethodArgs<'hello(string)string'>, params?: AppClientCallCoreParams & CoreAppCallArgs) {
+        promiseChain = promiseChain.then(() => client.hello(args, {...params, sendParams: {...params?.sendParams, skipSending: true, atc}}))
+        resultMappers.push(undefined)
+        return this
+      },
+      helloWorldCheck(args: MethodArgs<'hello_world_check(string)void'>, params?: AppClientCallCoreParams & CoreAppCallArgs) {
+        promiseChain = promiseChain.then(() => client.helloWorldCheck(args, {...params, sendParams: {...params?.sendParams, skipSending: true, atc}}))
+        resultMappers.push(undefined)
+        return this
+      },
+      get update() {
+        const $this = this
+        return {
+          bare(args?: BareCallArgs & AppClientCallCoreParams & AppClientCompilationParams & CoreAppCallArgs) {
+            promiseChain = promiseChain.then(() => client.update.bare({...args, sendParams: {...args?.sendParams, skipSending: true, atc}}))
+            resultMappers.push(undefined)
+            return $this
+          },
+        }
+      },
+      get delete() {
+        const $this = this
+        return {
+          bare(args?: BareCallArgs & AppClientCallCoreParams & CoreAppCallArgs) {
+            promiseChain = promiseChain.then(() => client.delete.bare({...args, sendParams: {...args?.sendParams, skipSending: true, atc}}))
+            resultMappers.push(undefined)
+            return $this
+          },
+        }
+      },
+      clearState(args?: BareCallArgs & AppClientCallCoreParams & CoreAppCallArgs) {
+        promiseChain = promiseChain.then(() => client.clearState({...args, sendParams: {...args?.sendParams, skipSending: true, atc}}))
+        resultMappers.push(undefined)
+        return this
+      },
+      async execute() {
+        await promiseChain
+        const result = await algokit.sendAtomicTransactionComposer({ atc, sendParams: {} }, client.algod)
+        return {
+          ...result,
+          returns: result.returns?.map((val, i) => val.returnValue !== undefined ? resultMappers[i]?.(val.returnValue) : undefined)
+        }
+      }
+    } as unknown as HelloWorldAppComposer
+  }
+}
+export type HelloWorldAppComposer<TSignatures extends [...(undefined | HelloWorldAppSig)[]] = []> = {
+  /**
+   * Calls the hello(string)string ABI method.
+   *
+   * Returns Hello, {name}
+   *
+   * @param args The arguments for the contract call
+   * @param params Any additional parameters for the call
+   * @returns The result of the call
```

Comment:
@robdmoore The returns value description here is "wrong" but this is probably more useful than saying "returns the composer object" each time

Reply:
I rekon something like:

> The transaction composer so you can fluently chain multiple calls or call `execute` to execute all queued up transactions